solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -1645,6 +1650,34 @@ public Set<Path> getAllowPaths() {
     return allowPaths;
   }
 
+  /**
+   * Checks that the given path is not protected (i.e. can accept writes). Protected paths generally
+   * contain library directories that Solr loads jars and classes from.
+   *
+   * @param pathToAssert path to check
+   * @throws SolrException if path is contained inside a library folder
+   */
+  public void assertPathNotProtected(Path pathToAssert) throws SolrException {
+    pathToAssert = pathToAssert.toAbsolutePath();
+    List<Path> protectedPathWithCores =
+        new ArrayList<>(protectedPaths.size() + solrCores.getNumAllCores());
+    protectedPathWithCores.addAll(protectedPaths);
+    solrCores.getCoreDescriptors().stream()
+        .map(CoreDescriptor::getInstanceDir)
+        .map(p -> p.resolve("lib").toAbsolutePath())
+        .forEach(protectedPathWithCores::add);
+
+    for (Path path : protectedPathWithCores) {
```

Comment:
Why the separation here with intermediate "protectedPathWithCores list; why not end with forEach and do the check?